test/httpboot/integration_test.rb

```diff
@@ -13,7 +12,7 @@ def app
   end
 
   def test_features
-    Proxy::DefaultModuleLoader.any_instance.expects(:load_configuration_file).with('httpboot.yml').returns(enabled: true, root_dir: '/var/lib/tftpboot')
+    Proxy::LegacyModuleLoader.any_instance.expects(:load_configuration_file).with('httpboot.yml').returns(enabled: true, root_dir: '/var/lib/tftpboot')
```

Comment:
This is a change which is weird, the only way of achieving what I want was to use `after_activation` block, however when that's present the class loading falls back to legacy module for some reason:

```ruby
def module_loader_class
    @module_loader_class ||= after_activation_blk.nil? ? ::Proxy::DefaultModuleLoader : ::Proxy::LegacyModuleLoader
  end
```

I don't feel like I want to refactor this in this PR.

Reply:
In https://github.com/theforeman/smart-proxy/pull/679 I implemented it using programmable settings.